cachi2/core/package_managers/bundler/parser.py

```diff
@@ -80,6 +80,32 @@ def download_to(self, deps_dir: RootedPath) -> None:
         download_binary_file(self.remote_location, fs_location)
 
 
+class GemPlatformSpecificDependency(GemDependency):
+    """
+    Represents a gem dependency built for a specific platform.
+
+    Attributes:
+        platform:     Platform for which the dependency was built.
+    """
+
+    platform: str
+
+    @cached_property
+    def remote_location(self) -> str:
+        """Return remote location to download this gem from."""
+        return f"{self.source}/downloads/{self.name}-{self.version}-{self.platform}.gem"
+
+    def download_to(self, deps_dir: RootedPath) -> None:
+        """Download represented gem to specified file system location."""
+        fs_location = deps_dir.join_within_root(
+            Path(f"{self.name}-{self.version}-{self.platform}.gem")
+        )
+        log.info(
+            "Downloading platform-specific gem %s-%s-%s", self.name, self.version, self.platform
+        )
+        download_binary_file(self.remote_location, fs_location)
+
+
```

Comment:
See my previous comment about duplication.

Reply:
We could rework it, but after I am done with sqlite3 (and potentially other interesting ways of storing versions).